src/lib/stepper/stepper.spec.ts

```diff
@@ -0,0 +1,574 @@
+      expect(stepperComponent.selectedIndex).toBe(0);
+    });
+
+    it('should change selected index on header click', () => {
+      let stepHeaders = fixture.debugElement.queryAll(By.css('.mat-vertical-stepper-header'));
+      checkSelectionChangeOnHeaderClick(stepperComponent, fixture, stepHeaders);
+
+    });
+
+    it('should set the "tablist" role on stepper', () => {
+      stepperEl = fixture.debugElement.query(By.css('md-vertical-stepper')).nativeElement;
+      expect(stepperEl.getAttribute('role')).toBe('tablist');
+    });
+
+    it('should set aria-expanded of content correctly', () => {
+      let stepContents = fixture.debugElement.queryAll(By.css(`.mat-vertical-stepper-content`));
+      checkExpandedContent(stepperComponent, fixture, stepContents);
+    });
+
+    it('should display the correct label', () => {
+      checkCorrectLabel(stepperComponent, fixture);
+    });
+
+    it('should go to next available step when the next button is clicked', () => {
+      checkNextStepperButton(stepperComponent, fixture);
+    });
+
+    it('should go to previous available step when the previous button is clicked', () => {
+      checkPreviousStepperButton(stepperComponent, fixture);
+    });
+
+    it('should set the correct step position for animation', () => {
+      checkStepPosition(stepperComponent, fixture);
+    });
+
+    it('should support keyboard events to move and select focus', () => {
+      let stepHeaders = fixture.debugElement.queryAll(By.css('.mat-vertical-stepper-header'));
+      checkKeyboardEvent(stepperComponent, fixture, stepHeaders);
+    });
+  });
+
+  describe('linear vertical stepper', () => {
+    let fixture: ComponentFixture<LinearMdVerticalStepperApp>;
+    let testComponent: LinearMdVerticalStepperApp;
+    let stepperComponent: MdVerticalStepper;
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(LinearMdVerticalStepperApp);
+      fixture.detectChanges();
+
+      testComponent = fixture.componentInstance;
+      stepperComponent = fixture.debugElement
+          .query(By.css('md-vertical-stepper')).componentInstance;
+    });
+
+    it('should have true linear attribute', () => {
+      expect(stepperComponent.linear).toBe(true);
+    });
+
+    it('should not move to next step if current step is not valid', () => {
+      expect(testComponent.oneGroup.get('oneCtrl')!.value).toBe('');
+      expect(testComponent.oneGroup.get('oneCtrl')!.valid).toBe(false);
+      expect(testComponent.oneGroup.valid).toBe(false);
+      expect(stepperComponent.selectedIndex).toBe(0);
+
+      let stepHeaderEl = fixture.debugElement
+          .queryAll(By.css('.mat-vertical-stepper-header'))[1].nativeElement;
+
+      checkLinearStepperValidity(stepHeaderEl, stepperComponent, testComponent, fixture);
+    });
+  });
+});
+
+function checkSelectionChangeOnHeaderClick(stepperComponent:
+                                               MdHorizontalStepper | MdVerticalStepper,
```

Comment:
just use common base class MdStepper? (here and other functions that take a stepper)